File: 프로그래머스/lv2/이하은/행렬의 곱셈.kt
class Solution {
    fun solution(A: Array<IntArray>, B: Array<IntArray>): Array<IntArray> {
        val N = A.size // A 행의 크기 
        val M = B[0].size // B 열의 크기 
        var answer = Array(N) { IntArray(M) { 0 } }
        
        // 행렬 곱셈의 조건
        // A 열의 크기 == B 행의 크기 
        for(i in 0 until N){
            for(j in 0 until M){
                for(k in 0 until A[0].size){
                    // A: 행 고정, 열 변화 
                    // B: 열 고정, 행 변화 
                    answer[i][j] += A[i][k] * B[k][j] 
                }
            }
        }
        
        return answer
    }
}

File: 프로그래머스/lv2/행렬의 곱셈.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12949

def solution(arr1, arr2):
    # arr1행과 arr2열 (arr1열과 arr2행은 같을테니)
    n = len(arr1)
    m = len(arr1[0])
    k = len(arr2[0])
    
    answer = [[0] * k for _ in range(n)]
    
    for i in range(n):# arr1의 각 행
        for l in range(k):# arr2의 각 열
            for j in range(m):# arr1의 열과 arr2의 행 (공통 크기)
                answer[i][l] += arr1[i][j] * arr2[j][l]
    
    return answer